core/src/main/scala/zio/jdbc/ZConnection.scala

```diff
@@ -86,8 +87,9 @@ final class ZConnection(private[jdbc] val connection: Connection) extends AnyVal
         i += 1
       }
 
-      f(statement)
-    }.tapErrorCause { cause =>
+      val close = ZIO.addFinalizer(ZIO.attemptBlocking(statement.close()).ignore)
+      (f(statement), close)
+    }.flatMap(vs => vs._2.as(vs._1)).tapErrorCause { cause =>
```

Comment:
This is strangely expressed because there's no need for a tuple. Maybe try something like:

```suggestion
     ZIO.addFinalizer(ZIO.attemptBlocking(statement.close()).ignore).as(f(statement))
    }.tapErrorCause { cause =>
```

Reply:
That was my original code, but `ZConnection.access` requires `Connection => A`, **not** `Connection => ZIO[R, E, A]`.
I think the following is the proper ZIO way to do it:
```suggestion
      statement
    }
    .flatMap(stmt => ZIO.addFinalizer(ZIO.attemptBlocking(stmt.close()).ignore).as(f(stmt)))
    .tapErrorCause { cause =>
```
(and I don't know how to add line numbers and start at the proper line, while still replying)

Reply:
We need to add `ZConnection.accessZIO`